scripts/publish/test_publish.py

```diff
@@ -368,6 +379,17 @@ def publish_project(target: str, uv: Path, client: httpx.Client):
     2. If we're using PyPI, uploading the same files again succeeds.
     3. Check URL works and reports the files as skipped.
     """
+    # If we're publishing to pyx, we need to give the httpx client
+    # access to an appropriate credential.
+    if target == "pyx-token":
+        client.headers.update(
+            {"Authorization": f"Bearer {os.environ['UV_TEST_PUBLISH_PYX_TOKEN']}"}
+        )
+    else:
+        # We reuse the same client for multiple targets, so make sure no
+        # credentials leak between targets.
+        client.headers.pop("Authorization", None)
```

Comment:
This feels like a pretty bad hack: this integration script has its own HTTP client that it uses to pull recent version numbers, so we need to give that client read access to pyx. Maybe there's a way we could do this all within uv itself, i.e. have uv provide the list of current versions?